app/src/main/java/ar/com/wolox/android/example/network/APIAdapter.java

```diff
@@ -0,0 +1,50 @@
+package ar.com.wolox.android.example.network;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import ar.com.wolox.android.example.model.APIClient;
+import ar.com.wolox.android.example.model.User;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+/**
+ * Adapter class for API
+ */
+public class APIAdapter {
+    private UserService userService2;
+
+    @Inject
+    public APIAdapter() {
+        userService2 = APIClient.getRetrofitClient().create(UserService.class);
+    }
+
+    /**
+     * @param mail - Mail entered by user
+     * @param password - Password entered by user
+     * @param listener - Invoked on login complete
+     */
+    public void getUserById(String mail, String password, OnLoginListener listener) {
```

Comment:
Well, I don't really like this implementation. Why are you sending `OnLoginListener` to presenter.